src/discord/config.go

```diff
@@ -0,0 +1,78 @@
+package discord
+
+import (
+	"github.com/devclub-iitd/DeployBot/src/helper"
+	"github.com/devclub-iitd/DeployBot/src/history"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	// postMessageHookURL is the webhook url for posting messages on discord
+	postMessageHookURL string
+	serverURL string
+)
+
+func init() {
+	postMessageHookURL = helper.Env("DISCORD_MESSAGE_WEBHOOK", "None")
+	if postMessageHookURL == "None" {
+		log.Fatal("DISCORD_MESSAGE_WEBHOOK not present in environment, Exiting")
+	}
+	serverURL = helper.Env("SERVER_URL", "https://listen.devclub.iitd.ac.in")
+}
+
+type Field struct {
+	Name	string	`json:"name"`
+	Value	string	`json:"value"`
+}
+
+type Embed struct {
+	Title string `json:"title"`
+	Fields []Field `json:"fields"`
+}
+
+type Message struct {
+	Content string `json:"content"`
+	Embeds []Embed `json:"embeds"`
+}
+
+func newMessage(a *history.ActionInstance, callbackID string) Message {
```

Comment:
I somehow don't like that package discord depends on packages history. 
My inital model was that controllers (is the root) and can depend on anything i.e. history, git, slack etc) and all others should only depend on helper and nothing else. 

Can you make these fields in the history package? instead of adding a dependency. Since the field type is a generic key value pair, you can put this in helpers.

Reply:
You can return a generic interface to keep thing much cleaner, like I did in https://github.com/devclub-iitd/DeployBot/blob/master/src/history/config.go#L100